docs/code/VBufferCareFeeding.md

```diff
@@ -142,129 +136,130 @@ using (IRowCursor cursor = dataview.GetRowCursor(col => col == 5))
+strongly resembles a `VBuffer<T>`, *except* instead of being `ReadOnlySpan`
+for values and indices they are `Span`, so, it is mutable.
+
+First, one creates the `VBufferEditor<T>` structure using one of the
+`VBufferEditor.Create` methods. You pass in a `VBuffer<T>` as a `ref`
+parameter. (For this purpose, it is useful to remember that a
+`default(VBuffer<T>)` is itself a completely valid, though completely empty,
+`VBuffer<T>`.) The editor is past that line considered to "own" the internal
+structure of the `VBuffer<T>` you passed in. (So, correct code should not
+continue to use that `VBuffer<T>` structure from that point onwards.)
+
+Then, one parties on the `Span<T>` for values and, if in a sparse application,
+the `Span<int>` for indices, on that editor, in whatever way is appropriate
+for the task.
+
+Last, one calls one of the `Commit` methods to get another `VBuffer`, with the
+values and indices accessible through the `VBuffer<T>` methods the same as
+those that had been set in the editor structure.
+
+Similar to how creating a `VBufferEditor<T>` out of a `VBuffer<T>` renders the
+passed in `VBuffer<T>` invalid, likewise, getting the `VBuffer<T>` out of the
+editor out of the `VBufferEditor<T>` through one of the commit methods renders
+the *editor* invalid. In both cases, "ownership" of the internal buffers is
+passed along to the successor structure, rendering the original structure
+invalid, in some sense.
+
+Internally, these buffers are backed by arrays that are reallocated *if
+needed* by the editor upon its creation, but reused if they are large enough.
+
+## Ownership of `VBuffer`
+
+Nonetheless, this does not mean that bugs are impossible, because the concept
+of buffer reuse does carry the implication that we must be quite clear about
+who "owns" a `VBuffer` at any given time, for which we have developed this
+convention:
+
+Unless clearly specified in documentation, a caller is assumed to always own a
+`VBuffer` as returned by `ref`. This means they can do whatever they like to
+it, like pass the same variable into some other getter, or modify its values.
 Indeed, this is quite common: normalizers in ML.NET get values from their
 source, then immediately scale the contents of `Values` appropriately. This
 would hardly be possible if the callee was considered to have some stake in
 that result.
 
+The `ValueGetter` indicated above is the most important example of this
+principle. By passing in an existing `VBuffer<T>` into that delegate by
+reference, they are giving the implementation control over it to use (or,
+reallocate) as necessary to store the resulting value. But, once the delegate
+returns with the value returned, the caller is now considered to own that
+buffer.
+
 There is a corollary on this point: because the caller owns any `VBuffer`,
-then you shouldn't do anything that irrevocably destroys their usefulness to
+then could should not do anything that irrevocably destroys its usefulness to
```

Comment:
Needs re-wording - `then could should not do anything`

Reply:
I meant to say "code" not "could." I think I'll post my commits now and not wait for you to finish, since this is the second time in a row we've independently noticed the same mistake in the docs, and I might save you some time potentially...